rust/operator-binary/src/superset_controller.rs

```diff
@@ -490,23 +489,30 @@ fn build_server_rolegroup_statefulset(
         add_authentication_volumes_and_volume_mounts(authentication_class, &mut cb, &mut pb)?;
     }
 
+    let webserver_timeout = node_config
+        .get(&PropertyNameKind::File(SUPERSET_CONFIG_FILENAME.to_string()))
+        .expect("Failed to get the superset config file from node config. It should be set by product config machinery")
+        .get(&SupersetConfigOptions::SupersetWebserverTimeout.to_string())
+        .expect("Failed to get webserver timeout from superset config file. It should be set by product config machinery");
```

Comment:
Do we really wanna stop the operator here? 
Lets assume we have a "bad" product config file (user created without the default for the timeout) and 2 clusters. One with and one without the timeout explicitly configured. One would work the other would make the operator panic. 
Do we want this?

Reply:
Yeah, please don't use `expect()` (or `unwrap()`) for errors that can be caused by user-controlled resources.

Reply:
My understanding is that the product-config will *always* put in the `SUPERSET_WEBSERVER_TIMEOUT` and even additionally validate that it's there - it's marked as
```
      roles:
        - name: "node"
          required: true
```
So i concluded the error can not be caused by any user input but only by a bug in our product-config machinery.
Was that conclusion wrong? (All tests were green btw)

Reply:
Well, iff the user is using our supplied product-config _and_ our product-config never changes this...

Reply:
Switched from the expect to a new Error variant. Hope that's ok